File: go/feast.go
package kiahk

// Feast is a calendar-resolved feast: a FeastRecord paired with the
// Gregorian date on which it falls for a particular year.
type Feast struct {
	ID            string
	Type          string            // "fixed" | "moveable"
	Category      string            // "major" | "minor"
	Names         map[string]string // locale → localized name
	GregorianDate GregorianDate
}

// Name returns the feast's localized name for the given locale.
// Supported locales: "en", "ar". Unknown locales return *UnsupportedLocaleError.
func (f Feast) Name(locale string) (string, error) {
	if name, ok := f.Names[locale]; ok {
		return name, nil
	}
	return "", &UnsupportedLocaleError{FeastID: f.ID, Locale: locale}
}

File: dart/lib/src/feast.dart
import 'errors.dart';
import 'gregorian_date.dart';

/// A calendar-resolved feast: a [FeastRecord]'s metadata paired with the
/// [GregorianDate] on which it falls for a particular year.
class Feast {
  final String id;
  final String type;
  final String category;
  final Map<String, String> names;
  final GregorianDate gregorianDate;

  const Feast({
    required this.id,
    required this.type,
    required this.category,
    required this.names,
    required this.gregorianDate,
  });

  /// Return the feast's localized name for [locale].
  /// Supported locales: 'en', 'ar'. Unknown locales throw [UnsupportedLocaleException].
  String name(String locale) {
    final n = names[locale];
    if (n == null) {
      throw UnsupportedLocaleException(id, locale);
    }
    return n;
  }
}
